libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -835,67 +842,509 @@ bool GCS_MAVLINK::handle_mission_item(mavlink_message_t *msg, AP_Mission &missio
+        if (gcs().out_of_time()) {
+            break;
+        }
+
+#if DEBUG_SEND_MESSAGE_TIMINGS
+        retry_deferred_body_start = AP_HAL::micros();
+#endif
+
+        // check if any "specially handled" messages should be sent out
+        {
+            const int8_t next = deferred_message_to_send_offset();
+            if (next != -1) {
+                if (!do_try_send_message(deferred_message[next].id)) {
+                    break;
+                }
+                deferred_message[next].last_sent_ms = AP_HAL::millis16();
+                _next_deferred_message_to_send = -1; // deferred_message_to_send will recalculate
+#if DEBUG_SEND_MESSAGE_TIMINGS
+                const uint32_t stop = AP_HAL::micros();
+                const uint32_t delta = stop - retry_deferred_body_start;
+                if (delta > try_send_message_stats.max_retry_deferred_body_us) {
+                    try_send_message_stats.max_retry_deferred_body_us = delta;
+                    try_send_message_stats.max_retry_deferred_body_type = 1;
+                }
+#endif
+                continue;
+            }
+        }
+
+        // check for any messages that the code has explicity sent
+        ap_message next = (ap_message)pushed_ap_message_ids.first_set();
+        if (next != no_message_to_send) {
+            if (!do_try_send_message(next)) {
+                break;
+            }
+            pushed_ap_message_ids.clear(next);
+#if DEBUG_SEND_MESSAGE_TIMINGS
+            const uint32_t stop = AP_HAL::micros();
+            const uint32_t delta = stop - retry_deferred_body_start;
+            if (delta > try_send_message_stats.max_retry_deferred_body_us) {
+                try_send_message_stats.max_retry_deferred_body_us = delta;
+                try_send_message_stats.max_retry_deferred_body_type = 2;
+            }
+#endif
+            continue;
+        }
+
+        next = next_deferred_message_to_send();
+        if (next != no_message_to_send) {
+            if (!do_try_send_message(next)) {
+                break;
+            }
+            bucket_message_ids_to_send.clear(next);
+            if (bucket_message_ids_to_send.first_set() == -1) {
+                // we sent everything in the bucket.  Reschedule it.
+                deferred_message_bucket[sending_bucket_id].last_sent_ms = AP_HAL::millis16();
+                find_next_bucket_to_send();
+            }
+#if DEBUG_SEND_MESSAGE_TIMINGS
+                const uint32_t stop = AP_HAL::micros();
+                const uint32_t delta = stop - retry_deferred_body_start;
+                if (delta > try_send_message_stats.max_retry_deferred_body_us) {
+                    try_send_message_stats.max_retry_deferred_body_us = delta;
+                    try_send_message_stats.max_retry_deferred_body_type = 3;
+                }
+#endif
+            continue;
+        }
+        break;
+    }
+#if DEBUG_SEND_MESSAGE_TIMINGS
+                const uint32_t stop = AP_HAL::micros();
```

Comment:
indenting is a few characters off although it's no big deal of course

Reply:
Fixed.